Fix: Navigation open submenu on click unavailable for page-list

AhmarZaidi opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

This PR fixes https://github.com/WordPress/gutenberg/issues/65513

## What?
The solution keeps the "open on click" toggle visible, allowing users to toggle it even if the `core/navigation-submenu` blocks are removed from a `core/navigation` block.

## Why?
This change addresses an issue where the "open on click" setting could not be adjusted if all submenus were removed. By keeping the toggle visible, users can manage the setting regardless of the presence of submenu blocks.

## How?
The implementation removes the `hasSubmenus` check before rendering the `Submenus` Toggle Control Group.

## Testing Instructions
1. Open a post or page in the editor.
2. Insert a `core/navigation` block.
3. Add a `core/navigation-submenu` block within it.
4. Enable the "open on click" setting.
5. Remove the `core/navigation-submenu` block.
6. Verify that the "open on click" toggle remains visible and can be manually adjusted.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast <!-- if applicable -->

https://github.com/user-attachments/assets/fd98fb42-ae6d-4ed3-9dd2-ea38ef2c477a

AhmarZaidi committed:
Reset open on click attribute in core/navigation block

AhmarZaidi committed:
Add empty inner blocks guard condition

AhmarZaidi committed:
Revert reset open on click attribute

AhmarZaidi committed:
Merge branch 'trunk' into fix/navigation-open-submenu-unavailable-for-page-list

AhmarZaidi committed:
Show submenus option in all cases for navigation block

carolinan commented:
Hi @AhmarZaidi do you still plan to work on this?

AhmarZaidi commented:
@carolinan I think it's ready to be reviewed. Let me know if any updates are required.

carolinan commented:
I don't believe it is resetting the **value** of option when the last submenu is removed, and I don't think it should be reset: But either I am misunderstanding the PR description, or it needs to be updated to cover what the PR does.

AhmarZaidi commented:
Hey @carolinan,
Right, initially the PR did reset the option but we changed [the approach](https://github.com/WordPress/gutenberg/issues/65513#issuecomment-2571794983) and I forgot to update the description 😅.

I've updated the description now.

carolinan reviewed (approved):
Thank you for the update.
Without this change, there is no way to know that there is a stored setting without opening the code editor mode. 
I believe this is the correct solution, but I am not comfortable merging this without a design review.
